http/middleware.go

```diff
@@ -0,0 +1,150 @@
+				return
+			}
+
+			token := strings.TrimPrefix(authorization, "Bearer ")
+			_, err := jwt.Decode(ctx, &jwt.DecodeParams{Token: token})
+			if err != nil {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			params := &AuthorizationParams{}
+			for _, opt := range opts {
+				err = opt(params)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
+			}
+			params.Token = token
+			claims, err := jwt.Verify(ctx, &params.VerifyParams)
+			if err != nil {
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
+			// Token was verified. Add the session claims to the request context
+			sessionCtx := clerk.ContextWithSessionClaims(r.Context(), claims)
+			next.ServeHTTP(w, r.WithContext(sessionCtx))
+		})
+	}
+}
+
+type AuthorizationParams struct {
+	jwt.VerifyParams
+}
+
+// AuthorizationOption is a functional parameter for configuring
+// authorization options.
+type AuthorizationOption func(*AuthorizationParams) error
+
+// AuthorizedParty sets the authorized parties that will be checked
+// against the azp JWT claim.
+func AuthorizedParty(parties ...string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.SetAuthorizedParties(parties...)
+		return nil
+	}
+}
+
+// CustomClaims allows to pass a type (e.g. struct), which will be populated with the token claims based on json tags.
+// You must pass a pointer for this option to work.
+func CustomClaims(claims any) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.CustomClaims = claims
+		return nil
+	}
+}
+
+// Leeway allows to set a custom leeway when comparing time values
+// for JWT verification.
+// The leeway gives some extra time to the token. That is, if the
+// token is expired, it will still be accepted for 'leeway' amount
+// of time.
+// This option accomodates for clock skew.
+func Leeway(leeway time.Duration) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.Leeway = leeway
+		return nil
+	}
+}
+
+// ProxyURL can be used to set the URL that proxies the Clerk Frontend
+// API. Useful for proxy based setups.
+// See https://clerk.com/docs/advanced-usage/using-proxies
+func ProxyURL(proxyURL string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.ProxyURL = clerk.String(proxyURL)
+		return nil
+	}
+}
+
+// Satellite can be used to signify that the authorization happens
+// on a satellite domain.
+// See https://clerk.com/docs/advanced-usage/satellite-domains
+func Satellite(isSatellite bool) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.IsSatellite = isSatellite
+		return nil
+	}
+}
+
+// JSONWebKey allows to provide a custom JSON Web Key based on which
+// the authorization JWT will be verified.
+func JSONWebKey(key string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		// From the Clerk docs: "Note that the JWT Verification key is not in
```

Comment:
🔧 Might be worth adding a unit test here, even for illustrating the use a bit better.

Reply:
My intention was to add an integration test for the options, but setting the tests up took more time than expected. I'm planning to get back to this once I find some time.